### PR TITLE
add mlego/m20 a numpad supporting quite few rp2040 boards, mainly the…

### PR DESCRIPTION
**Pull request by alinelena**
… ones which are small variations on the original pinout

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/mlego/m20/m20.c [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    // Host Keyboard LED Status
+    led_t led_state = host_keyboard_led_state();
+    oled_write_P(led_state.num_lock ? PSTR("Lower ") : PSTR("    "), false);
+    oled_write_P(led_state.scroll_lock ? PSTR("Raise ") : PSTR("    "), false);
+#ifdef WPM_ENABLE
+    oled_write_P(PSTR("\nwpm: "), false);
+    uint8_t wpm = get_current_wpm();
+    oled_write_P(wpm != 0 ? get_u8_str(wpm,' ') : PSTR("   "), false);
+#endif
+}
```
Comment (waffle87) [inline keyboards/mlego/m20/m20.c:117]: This looks like it's all keymap-specific code? Anything that depends on things like layers or that not every user may want should be moved to a custom keymap.

File: keyboards/mlego/m20/rev1/info.json [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    },
+    "tapping": {
+        "toggle": 2
+    },
+    "url": "https://gitlab.com/m-lego/m20",
+    "usb": {
+        "device_version": "0.0.1",
+        "pid": "0x6460",
+        "vid": "0xBABA"
+    },
```
Comment (waffle87) [inline keyboards/mlego/m20/rev1/info.json:69]:
```suggestion
    },
    "community_layouts": ["ortho_5x4"],
```

File: keyboards/mlego/m20/rev1/rules.mk [hunk 3]
```diff
@@ -0,0 +1,2 @@
+WS2812_DRIVER = vendor        #
+OLED_DRIVER = SSD1306      # Enable Support for SSD1306 or SH1106 OLED Displays; Communicating over I2C
```
Comment (waffle87) [inline keyboards/mlego/m20/rev1/rules.mk:2]:
```suggestion
WS2812_DRIVER = vendor
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/mlego/m20/m20.c (hunk 1 above)
Comment (alinelena) [reply keyboards/mlego/m20/m20.c:117]: I am in two minds about this and over years I was given various suggestions. I can move it all in the keymap, but if I want to share the code between keymaps where do I put it?

Comment (alinelena) [reply keyboards/mlego/m20/m20.c:117]: happy to move it in the userspace if that is ok

Comment (alinelena) [reply keyboards/mlego/m20/m20.c]: done, hope properly

File: keyboards/mlego/m20/rev1/config.h [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+#define LED_LWR_PIN GP25
+#define LED_RSE_PIN GP14
+
+#define RP2040_BOOTLOADER_DOUBLE_TAP_RESET              // Activates the double-tap behavior
+#define RP2040_BOOTLOADER_DOUBLE_TAP_RESET_TIMEOUT 200U // Timeout window in ms in which the double tap can occur.
+#define RP2040_BOOTLOADER_DOUBLE_TAP_RESET_LED_MASK 0U
+
+#ifdef OLED_ENABLE
+#    define OLED_DISPLAY_128X32
```
Comment (drashna) [inline keyboards/mlego/m20/rev1/config.h:14]: Not an actual preset.  Is the default without any size set. 
```suggestion
```

**Pull request closed without merging**